File: 01 arrays/exercise - longestWord.js

// Longest Word
// https://coderbyte.com/information/Longest%20Word

var longestWord = function(text) {
  if (!text || text.length == 0)
    return undefined;

  var arr = text.match(/(\w+)/gi);

  arr  = arr.sort(function (a, b){
    return b.length - a.length;
  });

  return arr[0];
}


console.log(longestWord("fun&!! time"))

File: 06 arrays/exercise - longestWord.py
import re

def longestWord(text):
  if (None == text or 0 == len(text)):
    return None

  arr = re.findall("(\w+)", text)

  arr = sorted(arr, key=lambda  itm: len(itm), reverse=True)
  
  if (len(arr) > 0):
    return arr[0]
  
  return None


longestWord ('fun## time')